design/ceph/object-store-zone.md

```diff
@@ -0,0 +1,183 @@
+# radosgw-admin period update --commit
+```
+
+In the other scenario, there are more than one zones in a zonegroup.
+
+Care must be taken when changing which zone is the master zone.
+
+Please read the following documentation before running the below commands: 
+
+https://docs.ceph.com/docs/master/radosgw/multisite/#changing-the-metadata-master-zone
+
+The following commands, run via toolboxes, remove the zone from the zonegroup first, then delete the zone.
+
+```bash
+# radosgw-admin zonegroup rm --rgw-zone=zone-z
+# radosgw-admin period update --commit
+# radosgw-admin zone rm --rgw-zone=zone-z
+# radosgw-admin period update --commit
+```
+### Changing the Master Zone through Toolboxes
+
+Similar to deleting zones, the Rook toolbox can also change the master zone in a zonegroup.
+
+```bash
+# radosgw-admin zone modify --rgw-zone=zone-z --master
+# radosgw-admin zonegroup modify --rgw-zonegroup=zonegroup-b --master
+# radosgw-admin period update --commit
+```
+
+## Object Store Zone CRD
+
+The object store zone settings are exposed to Rook as a Custom Resource Definition (CRD). The CRD is the Kubernetes-native means by which the Rook operator can watch for new resources.
+
+The name of the resource provided in the `metadata` section becomes the name of the zone.
+
+The following variables can be configured in the object-store-zone resource.
+
+- `zonegroup`: The zonegroup named in the `zonegroup` section of the object-store-realm resource the zone is a part of.
+- `isMaster`: This boolean value signifies whether the zone is master or not. Only one zone can be master. Any object-store-zone resource created as master after a master zone has been created, will not be created.
+
+```yaml
+apiVersion: ceph.rook.io/v1alpha1
+kind: CephObjectStorezone
+metadata:
+  name: zone-b
+  namespace: rook-ceph
+spec:
+  realm: my-realm
+  isMaster: True
+```
+
+### Pools
+
+The pools are the backing data store for the object stores in the zone and are created with specific names to be private to a zone.
+As long as the `zone` config option is specified in the object-store's config, the object-store will use pools defined in the object-store-zone's configuration.
+Pools can be configured with all of the settings that can be specified in the [Pool CRD](/Documentation/ceph-pool-crd.md).
+The underlying schema for pools defined by a pool CRD is the same as the schema under the `metadataPool` and `dataPool` elements of the object store CRD.
+All metadata pools are created with the same settings, while the data pool can be created with independent settings.
+The metadata pools must use replication, while the data pool can use replication or erasure coding.
+
+`preservePoolsOnDelete`, if it is set to 'true', ensures the pools used to support the zone will remain when the object-store-zone will be deleted. This is a security measure to avoid accidental loss of data. It is set to 'true' by default. If not specified is also deemed as 'true'.
```

Comment:
why do we need that if we don't support deletion at this phase?

Reply:
The resource itself could be deleted, it's just that the zone the resource represents wouldn't be deleted. Since the ceph-pool resources would be deleted since they are referenced in the CRD, we should make sure this is configured.